memory/pmem_dt_check.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2020 IBM
+# Author: Harish <harish@linux.vnet.ibm.com>
+
+"""
+Ndctl user space tooling for Linux, which handles NVDIMM devices.
+This Suite works with various options of ndctl on a NVDIMM device.
+"""
+
+import os
+
+import avocado
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import archive
+from avocado.utils import distro
+from avocado.utils import build
+from avocado.utils import genio
+from avocado.utils import pmem
+from avocado.utils import cpu
+from avocado.utils.software_manager import SoftwareManager
+
+
+class NdctlDeviceTreeCheck(Test):
+    """
+    Ndctl Test to check numa associativity for pmem devices from device tree
+    """
+    @staticmethod
+    def get_hex_list(filename):
+        out = process.system_output('lsprop %s' %
+                                    filename, shell=True).decode()
+        values = "".join(out.replace('\t\t', '').split("\n")
+                         [1:]).strip().split()
+        return values
+
+    def get_ref_point_index(self):
+        val = self.get_hex_list(
+            "/proc/device-tree/rtas/ibm,associativity-reference-points")
+        return int(val[0], 16)
+
+    def get_node_id_from_associativity(self, assoc_ref_index, filename):
+        values = self.get_hex_list(filename)
+        return int(values[assoc_ref_index], 16)
```

Comment:
Suggestion: would it not be better if decimal conversion is done in `get_hex_list()` itself ?
You could have a default value to the list index as 0, and override it when calling from `get_node_id_from_associativity()` with `assoc_ref_index`

Reply:
It is possible, but I would like to keep this way for two reasons.
1. better readability, so that it is straightforward to understand what is used for what
2. In future, `get_hex_list` may be used to get values which does not need any decimal conversion.

Reply:
okay, agree !!